src/main/java/bio/terra/cloudfunctions/proto/ProtoApp.java

```diff
@@ -0,0 +1,30 @@
+package bio.terra.cloudfunctions.proto;
+
+import bio.terra.cloudevents.CloudStorageEventType;
+import bio.terra.cloudevents.GCSEvent;
+import bio.terra.cloudfunctions.common.App;
+import java.util.logging.Logger;
+
+/** This class is a subclass of App with a custom implementation of the process() logic. */
+public class ProtoApp extends App {
+  private static final Logger logger = Logger.getLogger(ProtoApp.class.getName());
+
+  public ProtoApp(CloudStorageEventType cloudStorageEventType, GCSEvent cloudStorageEvent) {
```

Comment:
I don't think ProtoApp or other App subclasses should use GCSEvent or other classes that are closely coupled to GCP. That way we can switch to K8s or Azure by just moving the App subclasses to new function entry points.

Reply:
Updated the `App` class to use the `FileUploadedMessage` pojo.